pkg/redshift/api/api_test.go

```diff
@@ -216,3 +216,51 @@ func Test_GetSecret(t *testing.T) {
 		t.Errorf("unexpected result: %v", cmp.Diff(expectedSecret, secret))
 	}
 }
+
+func Test_GetCluster(t *testing.T) {
+	c := &API{ManagementClient: &redshiftclientmock.MockRedshiftClient{Clusters: []string{"foo"}}}
+	cluster, err := c.Cluster(sqlds.Options{"clusterIdentifier": "foo"})
+	if err != nil {
+		t.Fatalf("unexpected error %v", err)
+	}
+	expectedCluster := &models.RedshiftCluster{
+		Endpoint: models.RedshiftEndpoint{ 
+			Address: "foo", 
+			Port: 123,
+		}, 
+		Database: "foo",
+	}
+	if !cmp.Equal(expectedCluster, cluster) {
+		t.Errorf("unexpected result: %v", cmp.Diff(expectedCluster, cluster))
+	}
+}
+
+func Test_GetCluster_Errors(t *testing.T) {
+	fooC := &API{ManagementClient: &redshiftclientmock.MockRedshiftClient{Clusters: []string{"foo"}}}
+
+	tests := []struct {
```

Comment:
extra points if you merge this test with the previous one :) (making `expectedCluster` and the error optional).

Reply:
Yes that was the plan! 
Still trying to figure out a test for the other errors too 😉